optaplanner-core/src/main/java/org/optaplanner/core/impl/partitionedsearch/DefaultPartitionedSearchPhase.java

```diff
@@ -146,6 +145,7 @@ public void solve(DefaultSolverScope<Solution_> solverScope) {
         } finally {
             // If a partition thread throws an Exception, it is propagated here
             // but the other partition thread won't finish any time soon, so we need to terminate them
+            // TODO move to catch block
```

Comment:
I remember doing manual tests around this, as not cleaning up threads can bring down a server to it's knees.